### PR TITLE
Gradle builds

### PR DESCRIPTION
**Pull request by billbindi**
Move away from the ant builds and use gradle builds now.

2 notable things *NOT* in this change:

1. The benchmark task from the ant build
2. Documentation update in the various readmes about how to get going with this. This is planned for a future update.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+repositories {
+    jcenter()
+}
+
+dependencies {
+    testCompile 'junit:junit:4.12'
+}
+
+// keep group/version info the same as from before
+group 'com.palantir.patches.sourceforge'
```
Comment (billbindi) [inline build.gradle:48]: This is the group that was used before, so I just kept it the same

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+version '3.0.3-p6'
```
Comment (billbindi) [inline build.gradle:49]: Intentionally hardcoded for now. Happy to swap over to git-version plugin, but would require doing some tagging and better git management.

**Comment by nathklei** [comment]
@billbindi just to confirm, this will run unit tests against java 8 and 11 in the build, right? If so that's great - confirms our off-heap impls work on both, at least in unit testing.

**Review by nathklei — APPROVED** [review]

**Comment by billbindi** [comment]
> @billbindi just to confirm, this will run unit tests against java 8 and 11 in the build, right? If so that's great - confirms our off-heap impls work on both, at least in unit testing.

Yup, it runs both!

**Pull request merged**